notebook/static/notebook/js/keyboardmanager.js

```diff
@@ -145,25 +175,27 @@ define([
         /**
          *  returning false from this will stop event propagation
          **/
-
         if (event.which === keycodes.esc) {
             // Intercept escape at highest level to avoid closing
             // websocket connection with firefox
             event.preventDefault();
         }
-        
-        if (!this.enabled) {
-            if (event.which === keycodes.esc) {
-                this.notebook.command_mode();
-                return false;
-            }
-            return true;
-        }
-        
-        if (this.mode === 'edit') {
-            return this.edit_shortcuts.call_handler(event);
-        } else if (this.mode === 'command') {
-            return this.command_shortcuts.call_handler(event);
+
+        if (this.view === 'tree'){
+          this.tree_shortcuts.call_handler(event);
+        }else{
+          if (!this.enabled) {
+              if (event.which === keycodes.esc) {
+                  this.notebook.command_mode();
+                  return false;
+              }
+              return true;
+          }
+          if (this.mode === 'edit') {
```

Comment:
This logic is only present in notebook view (tree view has no modes); move into the else branch on the `this.view === 'tree'` conditional